validation/security-configs_test.go

```diff
@@ -0,0 +1,167 @@
+package validation
+
+import (
+	"context"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/ethereum-optimism/optimism/op-bindings/bindings"
+	"github.com/ethereum-optimism/optimism/op-service/retry"
+	. "github.com/ethereum-optimism/superchain-registry/superchain"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func testSecurityConfigOfChain(t *testing.T, chainID uint64) {
+	rpcEndpoint := Superchains[OPChains[chainID].Superchain].Config.L1.PublicRPC
+	require.NotEmpty(t, rpcEndpoint, "no rpc specified")
+
+	client, err := ethclient.Dial(rpcEndpoint)
+	require.NoErrorf(t, err, "could not dial rpc endpoint %s", rpcEndpoint)
+
+	type resolution struct {
+		name                     string
+		method                   string
+		shouldResolveToAddressOf string
+	}
+
+	contractCallResolutions := []resolution{
+		{"AddressManager", "owner()", "ProxyAdmin"},
+		{"SystemConfigProxy", "owner()", "SystemConfigOwner"},
+		{"ProxyAdmin", "owner()", "ProxyAdminOwner"},
+		{"L1CrossDomainMessengerProxy", "PORTAL()", "OptimismPortalProxy"},
+		{"L1ERC721BridgeProxy", "admin()", "ProxyAdmin"},
+		{"L1ERC721BridgeProxy", "messenger()", "L1CrossDomainMessengerProxy"},
+		{"L1StandardBridgeProxy", "getOwner()", "ProxyAdmin"},
+		{"L1StandardBridgeProxy", "messenger()", "L1CrossDomainMessengerProxy"},
+		{"OptimismMintableERC20FactoryProxy", "admin()", "ProxyAdmin"},
+		{"OptimismMintableERC20FactoryProxy", "BRIDGE()", "L1StandardBridgeProxy"},
+		{"OptimismPortalProxy", "admin()", "ProxyAdmin"},
+		{"ProxyAdmin", "owner()", "ProxyAdminOwner"},
+		{"ProxyAdmin", "addressManager()", "AddressManager"},
+		{"SystemConfigProxy", "admin()", "ProxyAdmin"},
+		{"SystemConfigProxy", "owner()", "SystemConfigOwner"},
```

Comment:
We already have an `Address` type for addresses, wondering if we want to create an `type AddressName string` and put some or most of these address names here as well into actual `const AddressName`s for reuse, maybe even inside the `superchain` package? Possibly in a follow-up, but we do copy some of these lots of time, like `ProxyAdmin` and it could just be in a const, avoiding typos. There could even be a package just for those consts, like `superchain/addrs`.

Reply:
I like this idea. I think it would be best to do in a followup, there is something similar already existing in the add-chain module that would need to be tidied up, and after starting on this it started to snowball a bit.